include/can/core/messages.hpp

```diff
@@ -41,19 +41,25 @@ struct BaseMessage {
  */
 template <MessageId MId>
 struct Empty : BaseMessage<MId> {
+    uint32_t message_index;
```

Comment:
Should we put the `message_index` member in `BaseMessage`? Everything inherits from it anyway. We could then even put a `parse_message_id` and `serialize_message_id` in it.

Reply:
spent a few hours trying to get this to work, the parsing and serializing work but it messes up the inheirtance for the non-empty child structs. we would have to change the instantiation of substructs everwhere and add constructors to them all
it's because it makes the BaseMessage a non-aggregate type and we'd have to do all that to get brace initalization working
I'll leave what I got in a branch but I think its better the way it is
For info this is similar to whats going on https://stackoverflow.com/questions/64770166/why-i-can-not-use-designated-initalizers-with-structs-that-are-not-aggregates